### PR TITLE
Add ability to disable operator through meshconfig

### LINKED ISSUE #7233: Add ability to disable the operator for a given meshconfig context
#### Current Behavior

Currently the support is added for disabling the operator at Meshery server boot time. There are no other nobs and dials for it as of now.
#### Desired Behavior

Add support to make it configurable from meshconfig per kubernetes context that Meshery detects.
#### Mockups

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by abdullah1308**
**Notes for Reviewers**

1. Operator can be disabled for a meshconfig context through setting an attribute `operator` to `disabled`
    image

2. `operator` attribute has no effect on a Docker context

This PR fixes #7233 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@Azanul fyi

File: mesheryctl/internal/cli/root/config/config_test.go
Comment (hexxdump) [inline mesheryctl/internal/cli/root/config/config_test.go:276]: it'd be good if we can change got/want to  received/expected

File: mesheryctl/pkg/utils/helpers.go
Comment (hexxdump) [inline mesheryctl/pkg/utils/helpers.go:958]: Why cant we use the below statement like the other env variables? 
```
        valueOverrides["env"] = map[string]interface{}{
			"DISABLE_OPERATOR": true,
		}
```

File: mesheryctl/internal/cli/root/config/config_test.go
Comment (abdullah1308) [reply mesheryctl/internal/cli/root/config/config_test.go:276]: @hexxdump The other tests follow the same got/want naming convention. It can be the same for consistency sake.

File: mesheryctl/pkg/utils/helpers.go
Comment (abdullah1308) [reply mesheryctl/pkg/utils/helpers.go:958]: @hexxdump It is written like this to prevent overriding of the PROVIDER value -

image

Comment (hexxdump) [reply mesheryctl/pkg/utils/helpers.go:958]: By looking at lines 933-951, that does not seem probable. As per your comment, components, image and provider must overwrite each other's values.

Comment (abdullah1308) [reply mesheryctl/pkg/utils/helpers.go:958]: @hexxdump In 933-951, the values are written to different keys hence overwriting doesn't occur. Image is written to `image` and provider is written to `env`. Whereas for assigning `PROVIDER` and `DISABLE_OPERATOR` the value should be the written to the same `env` key.

Comment (abdullah1308) [reply mesheryctl/pkg/utils/helpers.go:958]: > Why cant we use the below statement like the other env variables?
> 
> ```
>         valueOverrides["env"] = map[string]interface{}{
> 			"DISABLE_OPERATOR": true,
> 		}
> ```

If it is written like above the lines 948-950 have no effect.

Comment (hexxdump) [reply mesheryctl/pkg/utils/helpers.go:958]: > @hexxdump In 933-951, the values are written to different keys hence overwriting doesn't occur. Image is written to `image` and provider is written to `env`. Whereas for assigning `PROVIDER` and `DISABLE_OPERATOR` the value should be the written to the same `env` key.

@abdullah1308 , got it! Thanks for the clarification.

**Review by hexxdump — APPROVED** [review]
LGTM

File: mesheryctl/pkg/utils/helpers.go
Comment (abdullah1308) [reply mesheryctl/pkg/utils/helpers.go:955]: Done

File: mesheryctl/internal/cli/root/config/config_test.go
Comment (abdullah1308) [reply mesheryctl/internal/cli/root/config/config_test.go:94]: `context` refers to a context in the meshconfig. Its meaning is self explanatory in the file. Changing the name of the variable here would make it inconsistent with many other tests.

**Comment by leecalcote** [comment]
😎

File: mesheryctl/pkg/utils/helpers.go
Comment (Azanul) [inline mesheryctl/pkg/utils/helpers.go:955]: @abdullah1308 You misunderstood me
```suggestion
	if ctx.GetDisableOperator() {
		if _, ok := valueOverrides["env"]; ok {
```

File: mesheryctl/internal/cli/root/config/config_test.go
Comment (Azanul) [reply mesheryctl/internal/cli/root/config/config_test.go:94]: Yes, you're right on both accounts, the meaning of context being self-explanatory (to people who have already worked on meshery) and changing it at just one place would introduce an inconsistency (that's why we should change it in the whole file).

There's a good (practice) reason to do it and no reason to not do it.

**Review by Azanul — DISMISSED** [review]

**Comment by Azanul** [comment]
@abdullah1308 conflicts

**Comment by acald-creator** [comment]
ci failed on go linting? can you check again?

**Comment by abdullah1308** [comment]
> ci failed on go linting? can you check again?

@acald-creator fixed

**Comment by abdullah1308** [comment]
@hexxdump any updates on this ?

**Comment by hexxdump** [comment]
> @hexxdump any updates on this ?

@abdullah1308, I have no further comments, LGTM!

**Review by hexxdump — APPROVED** [review]

**Review by alphaX86 — APPROVED** [review]

**Review by Chadha93 — APPROVED** [review]
Thank you for the review, Maintainers.

**Pull request merged**
